File: app/lib/transporter.js

const nodemailer = require('nodemailer');
const config = require('./config');
const markdown = require('nodemailer-markdown').markdown;

// Setup mailer
const transporterConfig = {
    host: config.smtpHost,
    port: config.smtpPort,
    secure: config.smtpSsl,
    auth: (config.smtpAuth && config.smtpUser && config.smtpPass ? {
        user: config.smtpUser,
        pass: config.smtpPass
    } : false),
    tls: {
        rejectUnauthorized: config.smtpSsl
    }
};

console.log(transporterConfig);

const transporter = nodemailer.createTransport(transporterConfig);

// Use Markdown
transporter.use('compile', markdown());

module.exports = transporter;

File: src/main/kotlin/jlelse/maily/lib/Transporter.kt
package jlelse.maily.lib

import jlelse.maily.require
import kotlin.js.json

object Transporter {
    private val nodemailer = require("nodemailer")
    private val markdown = require("nodemailer-markdown").markdown

    private val transporterConfig = json(
            "host" to Config.smtpHost,
            "port" to Config.smtpPort,
            "secure" to Config.smtpSsl,
            "auth" to if (Config.smtpAuth && Config.smtpUser != null && Config.smtpPass != null) {
                json(
                        "user" to Config.smtpUser,
                        "pass" to Config.smtpPass
                )
            } else false,
            "tls" to json(
                    "rejectUnauthorized" to Config.smtpSsl
            )
    )
    val transporter: dynamic = nodemailer.createTransport(transporterConfig)

    init {
        console.log(transporterConfig)
        transporter.use("compile", markdown())
    }
}
